### ISSUE TITLE
[C++] Question : Model Save&Load / Restore

### ISSUE THREAD

**Issue by Eomtaesan** [Microsoft/CNTK#3386]
I saw the example source code about "SaveAndReload" function in "Common.h" file.
(https://github.com/Microsoft/CNTK/blob/987b22a8350211cb4c44278951857af1289c3666/Tests/EndToEndTests/CNTKv2Library/Common/Common.h)

The function "SaveAndReload" use combine method.

My questions are

1. why the function use "combine"? (I think that the output FunctionPtr is only needed to save model.)

2. what is difference between Function::Load and Function::Restore (not restore from check point)

The description of "Load" function is "Load a Function from a model file", and
the description of "Restore" function is "Restore the models parameters (in-place) from a model file".

Then to load model from a model file, I think, this following method is needed.

	void load ( void )
	{
		std::wstring path = L"network.model"

		result = CNTK::Function::Load ( path , device );
		input = result->Arguments ( ) [ 0 ];
		label = CNTK::InputVariable ( { nOutput } , false , CNTK::DataType::Float , true , L"label" );
		output = CNTK::Softmax ( result , L"output" );
		loss = CNTK::CrossEntropyWithSoftmax ( result , label , L"loss" );
		error = CNTK::ClassificationError ( result , label , 1 , L"error" );
		auto learner = CNTK::AdamLearner ( result->Parameters ( ) , CNTK::TrainingParameterPerSampleSchedule ( LearningRate ) , CNTK::MomentumSchedule ( 0.9 ) );
		trainer = CNTK::CreateTrainer ( result , loss , error , { learner } );
		result->Restore ( path );

		return;
	}

Is it right way?
I wonder The "Load" function include "Restore" or not.

**Comment by Eomtaesan** [comment]
I have failed learning.
I thought the reason is model loading failure, but that is correct.
Now I find the other reason that is about input format
I used CHW format, but this issue https://github.com/Microsoft/CNTK/issues/2723 say "C++ uses WHC format".

So close this

**Comment by bencherian** [comment]
`Restore` simply overwrites the parameter values in an existing `Function` object. `Load` creates a new `Function` object.

For example, in your case, the `result->Restore( path )` call is entirely redundant. One plausible use of restore would be to reset the model. For example, if you ran the trainer for a few minibatches, realized there was some problem and wanted to go back to using the original parameters without having to recreate all of the other objects, you would want to use `Restore`. If you were to say `result = CNTK::Function::Load(...)`, you would create a new `Function` object which `result` would refer to while all the other objects would still use the `Function` referred to by the original `result`.
